### ISSUE TITLE
PHP Warning:  A non-numeric value encountered + PHP Notice: A non well formed numeric value encountered

### ISSUE THREAD

**Issue by seb86** [co-cart/co-cart#285]
There are two PHP issues currently unsolved. One is a warning and the other just a notice. Both are very similar.

Accessing the WordPress dashboard causes a cron-job to have the warning.

```php
PHP Warning:  A non-numeric value encountered in /wp-includes/cron.php on line 849
```

The PHP notice apparently is caused during plugin activation but I have not been able to replicate that.

There are a few sites that run plugin tests of their own for plugins hosted on the WordPress plugin directory and CoCart happens to be one that they have tested and this notice fails the test.

```php
PHP Notice: A non well formed numeric value encountered in /wp-includes/functions.php on line 471
```

I would like to point out that both tests have not installed the required plugin that is WooCommerce. So that is more than likely to be the reason behind it the PHP notice.

Both test sites run different PHP versions but both tested using the following:

* WordPress version: 5.8.1
* MySQL version: 8.0.21
* PHP memory limit: 256M

## Test sites

* [PluginTests](https://plugintests.com/plugins/wporg/cart-rest-api-for-woocommerce/latest)
* [WP Hive](https://wphive.com/plugins/cart-rest-api-for-woocommerce/)

If your testing CoCart I'd appreciate any feedback to help find the cause of these PHP issues or a contribution to solve them.

**Comment by seb86** [comment]
A similar error that was reported for the return of the item weight could be a more definitive reason why this error occurred.

The REST API does now run in the WordPress dashboard even when not all endpoints are accessed directly due to all the changes made for Gutenberg and Widgets in the core of WordPress not to mention any WooCommerce pages and CoCart pages.

Seeing an update of tests from both these sites once the next CoCart patch is released would be helpful and hope the tests pass. If not then this error still lies elsewhere with no definitive location to the source.

**Comment by seb86** [comment]
Closing for now as there has not been any further reports.

**Comment by seb86** [comment]
Finally found the issue. A function required is not able to be called if WooCommerce is not available. Solved with issue #341
